File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
function height(root, obj) {
    
    if (root == null) {
        return 0
    }
    if (Math.abs(height(root.left,obj) - height(root.right,obj)) > 1) {
       obj.val = false;   // Update obj.val to false if the tree is unbalanced
        return null;
       
        // obj.val = false;
        return null

    }
    return (1 + Math.max(height(root.left,obj), height(root.right,obj)))
}
var isBalanced = function (root) {


    var obj = { val: true }
    var h1 = height(root, obj)

    return obj.val







};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int finheight(TreeNode *root, bool &flag){
        if(root==NULL){
            return 0;
        }
        if(abs(finheight(root->left,flag)- finheight(root->right,flag))>1){
            
            flag = false;
            return 123345;
        }
        return (1+max(finheight(root->left,flag), finheight(root->right,flag)) );
    }
    bool isBalanced(TreeNode* root) {

        if(root==NULL){
            return true;
        }
        bool flag = true;
        int h = finheight(root,flag);
        return flag;
    }
};